Ah.get tree images

huang0h opened the pull request:
## Why

Resolves #198 
Resolves #183 

Adds to the get site route to also retrieve user-uploaded images for the site

<!-- What benefit does this bring to the end user? Or, what benefit does this bring to developers working in the codebase? -->

## This PR

- Adds the `SiteEntryImage` DTO to represent site image information
- Update `getSite()` to also retrieve site images
- Assigned default images to tree species
- Updated import router to import species' default images

<!-- Describe the changes required and any implementation choices you made to give context to reviewers. -->

## Verification Steps

- Verified a site with images will correctly fetch only approved user-uploaded images
- Verified a site without images, but whose tree has a default image, fetches said default image
- Verified a site without images nor default image returns an empty list
- Verified site images with anonymous user returns "Anonymous" instead of the uploader's username

<!-- What steps did you take to verify your changes work? These should be clear enough for someone to be able to clone the branch and follow the steps themselves. -->

huang0h commented:
As the default images for tree species live in the table `tree_species`, I updated the import router so you can import which species get which images. You can call the route `POST api/v1/protected/import/tree_species` with [this json file](https://s3.us-east-2.amazonaws.com/sftt.import.data/tree_species.json) as its body (you might need to clear the `tree_species` table first)

CerberusLatrans commented:
> As the default images for tree species live in the table `tree_species`, I updated the import router so you can import which species get which images. You can call the route `POST api/v1/protected/import/tree_species` with [this json file](https://s3.us-east-2.amazonaws.com/sftt.import.data/tree_species.json) as its body (you might need to clear the `tree_species` table first)

I don't think I have access to see the json:
<img width="799" alt="Screenshot 2023-07-09 at 4 27 44 PM" src="https://github.com/Code-4-Community/speak-for-the-trees-backend-v2/assets/52577512/3294b7c1-07b1-4be0-9bfe-bf05c59a39a6">

CerberusLatrans commented on persist/src/main/resources/db/migration/V4_8__AddTreeSpeciesDefaultImage.sql:
Is 500 characters ensured to be enough for any tree URL? Not sure if there is some AWS url length specification for this

CerberusLatrans replied:
Also, if the default images are also being stored in AWS do you think it makes sense for them to have an image ID too?

CerberusLatrans commented on service/src/main/java/com/codeforcommunity/processor/SiteProcessorImpl.java:
Looks good! Do we have any system in place for actually approving user-submitted images?

huang0h commented:
> > As the default images for tree species live in the table `tree_species`, I updated the import router so you can import which species get which images. You can call the route `POST api/v1/protected/import/tree_species` with [this json file](https://s3.us-east-2.amazonaws.com/sftt.import.data/tree_species.json) as its body (you might need to clear the `tree_species` table first)
> 
> I don't think I have access to see the json: <img alt="Screenshot 2023-07-09 at 4 27 44 PM" width="799" src="https://user-images.githubusercontent.com/52577512/252167528-3294b7c1-07b1-4be0-9bfe-bf05c59a39a6.png">

Just sent the file to you on slack (and updated the link to take you to the object page instead) - sorry about that!

chromium-52 commented on service/src/main/java/com/codeforcommunity/processor/SiteProcessorImpl.java:
```diff
@@ -163,14 +172,55 @@ record -> {
                   record.getBiocharAdded(),
-                  record.getLastEditedUser());
+                  record.getLastEditedUser(),
+                  getSiteEntryImages(record.getId(), record.getCommonName()));
```
⛏️ maybe move this to line 137 because it's not a Cambridge field

chromium-52 reviewed (approved):
👍 lgtm! Nice work!
